tests/chainer_tests/functions_tests/rnn_tests/test_function_n_step_rnn.py

```diff
@@ -4,569 +4,257 @@
-    def test_inconsistent_input_size_gpu_cudnn_always(self):
-        self.check_inconsistent_input_size_gpu('always')
-
-    @attr.gpu
-    def test_inconsistent_input_size_gpu_cudnn_never(self):
-        self.check_inconsistent_input_size_gpu('never')
-
-
-@testing.parameterize(*testing.product({
-    'activation': ['tanh', 'relu']
-}))
-class TestNStepBiRNN(unittest.TestCase):
-
-    batches = [3, 2, 1]
-    length = len(batches)
-    in_size = 3
-    out_size = 2
-    n_layers = 2
-    dropout = 0.0
+        self.check_forward_options.update({
+            'rtol': 1e-2, 'atol': 1e-2})
+        self.check_backward_options.update({
+            'rtol': 1e-2, 'atol': 1e-2})
+        self.check_double_backward_options.update({
+            'rtol': 5e-3, 'atol': 5e-2})
+        self.skip_double_backward_test = True
+
+    def generate_inputs(self):
+        h_shape = (self.n_layers, self.batches[0], self.hidden_size)
+        dtype = numpy.float32
+
+        h = numpy.random.uniform(-1, 1, h_shape).astype(dtype)
+        in_size = self.input_size
+        out_size = self.hidden_size
+        xs = [numpy.random.uniform(-1, 1, (self.batches[b], in_size)).astype(dtype)
+              for b in range(len(self.batches))]
+
+        def w_in(i, j):
+            return in_size if i == 0 and j < 1 else out_size
+
+        inputs = []
+        inputs.append(h)
+        for i in range(len(self.batches)):
+            inputs.append(xs[i])
+        for n in range(self.n_layers):
+            for i in range(2):
+                inputs.append(numpy.random.uniform(-1, 1,
+                                                   (out_size, w_in(n, i))).astype(dtype))
+            for i in range(2):
+                inputs.append(numpy.random.uniform(-1, 1,
+                                                   (out_size,)).astype(dtype))
+        return tuple(inputs)
+
+    def process_inputs(self, inputs):
+        h = inputs[0]
+
+        xs = inputs[1:1 + len(self.batches)]
+        ws = []
+        bs = []
+        index = 1 + len(self.batches)
+        for n in range(self.n_layers):
+            ws.append(inputs[index: index + 2])
+            bs.append(inputs[index + 2: index + 4])
+            index += 4
+
+        return h, ws, bs, xs
+
+    def forward(self, inputs, device):
+        h, ws, bs, xs = self.process_inputs(inputs)
+        # For some reason even though only float32 arrays are created in generate_inputs(),
+        # arrays coming as input here are of type float32 and float64
+        if h.array.dtype == numpy.float64:
+            raise unittest.SkipTest('float64 not supported')
+        out = F.n_step_rnn(self.n_layers, 0.0, h, ws, bs, xs)
+        rets = []
+        rets.append(out[0])
+        for i in range(len(out[1])):
+            rets.append(out[1][i])
+        return tuple(rets)
+
+    def forward_expected(self, inputs):
+        h, ws, bs, xs = self.process_inputs(inputs)
+        with chainer.using_config('use_ideep', 'never'):
```

Comment:
Is this needed?

Reply:
Actually, I was following `test_convolution_2d.py` (https://github.com/chainer/chainer/blob/e2e6eaa90f4f01d905767b69503d56629b350055/tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py#L110), that's why I had added this. I will remove it.